### DISCUSSION TITLE
Does getting this setup break Google's ToS?

### DISCUSSION THREAD

**Discussion by NovaViper** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#192] [Q&A]
Hi, I want to really try and set this up on my laptop. However, I found this method from [this article](https://9to5google.com/2021/02/16/hands-on-with-the-unofficial-google-assistant-desktop-client-video/).. but it states that you could lose your Google Account by doing this. Is this true? And why would it break ToS?

**Comment by Melvin-Abraham** [comment]
@NovaViper I am not sure how this would break Google's Terms of Service as far as setting up the Assistant is concerned. I have seen many use similar ways to get Google Assistant on their PC/laptop via Python way before this app was released. On top of that, for anyone wanting to try out the APIs, this is pretty much the way to do it.

**PS:** If you are still a bit hesitant, you may use a Gmail account that you do not use primarily. Also, if anyone has any ideas regarding why this might break Google's ToS, please do leave a comment in this discussion.

**Reply by luximus-hunter** [reply]
It's probably too high risk high reward, but you can try to ask Google directly. But that's probably too risky 😬

**Reply by NovaViper** [reply]
I don't really get how it would break ToS, Tasker directly uses Google's APIs in order to perform certain actions too (notably the Google Voice API for the WaveNet voice functionality) and I've had that setup for years.. I feel like if they can do it, then it should be safe to do this.

Also, there was a very similar work around in order to get Google Assistant running on Windows 10 a couple of years back (using the same API), though it all had to be ran in terminal

**Reply by luximus-hunter** [reply]
Then i guess its alright, Its just use of the API then.

**Reply by Harsharma2308** [reply]
Is there a specific limit to the api calls that we should make? Wondering if it is too many API calls which breaks TOS

**Comment by obed9057** [comment]
if your asking if this goes against the google laws of coding it does not because this API was designed by google but made compatible and updated by @Melvin-Abraham
